src/Server.php

```diff
@@ -226,7 +229,24 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
         );
 
         $callback = $this->callback;
-        $callback($request, $response);
+
+        $promise = $callback($request);
+        if (!$promise instanceof Promise) {
+            $promise = new Promise(function ($resolve, $reject) use ($promise) {
+                $resolve($promise);
+            });
+        }
```

Comment:
The problem with `$promise = \React\Promise\resolve($callback($request));
` is that would have to use `try-catch`. With my lines the Exception would be handled by the `Promise` and will be recognized as rejected. But thanks for the line didn't knew it ;)